types/mongoose/index.d.ts

```diff
@@ -2551,6 +2551,7 @@ declare module "mongoose" {
      * @param fn optional callback
      */
     save(fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
+    save(options: SaveOptions, fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
```

Comment:
@fabiob One more thing. Switch these two lines and put `options?: SaveOptions` like before.